tests/cbmc/proofs/s2n_socket_read_snapshot/s2n_socket_read_snapshot_harness.c

```diff
@@ -0,0 +1,26 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * @file s2n_socket_read_snapshot_harness.c
+ * @brief Implements the proof harness for s2n_socket_read_snapshot function.
+ */
+
+#include <utils/s2n_socket.h>
+
+void s2n_socket_read_snapshot_harness()
+{
+  struct s2n_socket_read_io_context *cbmc_allocate_s2n_socket_read_io_context();
+  /* Non-deterministic inputs. */
```

Comment:
(Nitpick - same for other harnesses) It should be
```c
/* Non-deterministic inputs. */
struct s2n_socket_read_io_context *cbmc_allocate_s2n_socket_read_io_context();
struct s2n_connection *s2n_connection = malloc(sizeof(*s2n_connection));
if (s2n_connection != NULL) {
    s2n_connection->recv_io_context = cbmc_allocate_s2n_socket_read_io_context();
}
```

Reply:
Why do we need the first line:

```
struct s2n_socket_read_io_context *cbmc_allocate_s2n_socket_read_io_context();
```